mono/utils/mono-state.h

```diff
@@ -0,0 +1,30 @@
+/**
+ * \file
+ * Support for interop with the Microsoft Error Reporting tool (header)
+ *
+ * Author:
+ *   Alexander Kyte (alkyte@microsoft.com)
+ *
+ * (C) 2018 Microsoft, Inc.
+ *
+ */
+#ifndef __MONO_UTILS_NATIVE_STATE__
+#define __MONO_UTILS_NATIVE_STATE__
+
+#include <config.h>
```

Comment:
Is config.h banned from .h files? I thought so.

Reply:
```git grep 'include.*config.h'  | grep '.h:'
libgc/Makefile.direct:if_mach: $(srcdir)/if_mach.c $(srcdir)/include/private/gcconfig.h
libgc/include/private/gc_priv.h:#   include "gcconfig.h"
mono/dis/push-pop.h:#include <config.h>
mono/eglib/glib.h:#pragma include_alias(<eglib-config.h>, <eglib-config.hw>)
mono/eglib/glib.h:#include <eglib-config.h>
mono/eglib/gmodule-win32-internals.h:#include <config.h>
mono/eglib/test/test.h:#include <config.h>
mono/metadata/abi-details.h:#include <config.h>
mono/metadata/cominterop-win32-internals.h:#include <config.h>
mono/metadata/console-io.h:#include <config.h>
mono/metadata/console-win32-internals.h:#include <config.h>
mono/metadata/coree-internals.h:#include <config.h>
mono/metadata/coree.h:#include <config.h>
mono/metadata/debug-mono-ppdb.h:#include <config.h>
mono/metadata/fdhandle.h:#include <config.h>
mono/metadata/file-mmap.h:#include <config.h>
mono/metadata/handle.h:#include <config.h>
mono/metadata/icall-internals.h:#include <config.h>
mono/metadata/icall-table.h:#include <config.h>
mono/metadata/icall-windows-internals.h:#include <config.h>
mono/metadata/marshal-internals.h:#include <config.h>
mono/metadata/marshal-windows-internals.h:#include <config.h>
mono/metadata/method-builder-ilgen-internals.h:#include "config.h"
mono/metadata/method-builder-ilgen.h:#include "config.h"
mono/metadata/method-builder-internals.h:#include "config.h"
mono/metadata/method-builder.h:#include "config.h"
mono/metadata/mono-config-dirs.h:#include <config.h>
mono/metadata/mono-security-windows-internals.h:#include <config.h>
mono/metadata/remoting.h:#include "config.h"
mono/metadata/sgen-bridge-internals.h:#include "config.h"
mono/metadata/sgen-mono-ilgen.h:#include "config.h"
mono/metadata/sysmath.h:#include <config.h>
mono/metadata/threadpool-io.h:#include <config.h>
mono/metadata/threadpool.h:#include <config.h>
mono/metadata/w32error.h:#include <config.h>
mono/metadata/w32event.h:#include <config.h>
mono/metadata/w32file-internals.h:#include <config.h>
mono/metadata/w32file-win32-internals.h:#include <config.h>
mono/metadata/w32file.h:#include <config.h>
mono/metadata/w32handle-namespace.h:#include <config.h>
mono/metadata/w32handle.h:#include <config.h>
mono/metadata/w32mutex.h:#include <config.h>
mono/metadata/w32process-internals.h:#include <config.h>
mono/metadata/w32process-unix-internals.h:#include <config.h>
mono/metadata/w32process-win32-internals.h:#include <config.h>
mono/metadata/w32process.h:#include <config.h>
mono/metadata/w32semaphore.h:#include <config.h>
mono/metadata/w32socket-internals.h:#include <config.h>
mono/metadata/w32socket.h:#include <config.h>
mono/mini/dwarfwriter.h:#include "config.h"
mono/mini/ee.h:#include <config.h>
mono/mini/image-writer.h:#include "config.h"
mono/mini/lldb.h:#include "config.h"
mono/mini/mini-windows.h:#include <config.h>
mono/mini/mini.h:#include "config.h"
mono/profiler/aot.h:#include <config.h>
mono/sgen/gc-internal-agnostic.h:#include <config.h>
mono/sgen/sgen-gc.h:#include "config.h"
mono/sgen/sgen-hash-table.h:#include "config.h"
mono/utils/atomic.h:#include "config.h"
mono/utils/checked-build.h:#include <config.h>
mono/utils/gc_wrapper.h:#include <config.h>
mono/utils/mach-support.h:#include "config.h"
mono/utils/mono-compiler.h:#include <config.h>
mono/utils/mono-complex.h:#include <config.h>
mono/utils/mono-coop-mutex.h:#include <config.h>
mono/utils/mono-coop-semaphore.h:#include <config.h>
mono/utils/mono-digest.h:#include <config.h>
mono/utils/mono-dl-windows-internals.h:#include <config.h>
mono/utils/mono-hwcap-vars.h:#include "config.h"
mono/utils/mono-hwcap.h:#include "config.h"
mono/utils/mono-io-portability.h:#include "config.h"
mono/utils/mono-lazy-init.h:#include <config.h>
mono/utils/mono-machine.h:#include "config.h"
mono/utils/mono-membar.h:#include <config.h>
mono/utils/mono-memory-model.h:#include <config.h>
mono/utils/mono-merp.h:#include <config.h>
mono/utils/mono-mmap-windows-internals.h:#include <config.h>
mono/utils/mono-os-mutex.h:#include <config.h>
mono/utils/mono-os-semaphore.h:#include <config.h>
mono/utils/mono-os-wait.h:#include <config.h>
mono/utils/mono-poll.h:#include <config.h>
mono/utils/mono-proclib-windows-internals.h:#include <config.h>
mono/utils/mono-publib.h: * We can't include config.h, so we use compiler-specific preprocessor
mono/utils/mono-rand-windows-internals.h:#include <config.h>
mono/utils/mono-sigcontext.h:#include <config.h>
mono/utils/mono-signal-handler.h:#include "config.h"
mono/utils/mono-threads-coop.h:#include <config.h>
mono/utils/mono-threads.h:#include <config.h>
mono/utils/mono-tls.h:#include <config.h>
mono/utils/networking.h:#include <config.h>
mono/utils/os-event.h:#include <config.h>
mono/utils/refcount.h:#include <config.h>
support/libm/complex.h:#include <config.h>
support/mph.h:#include <config.h>
```

Reply:
I think there was a PR/commit that deliberately reduced it. Or maybe it was just being sure config.h was first. Or maybe it was removing from public headers.

Reply:
https://github.com/mono/mono/commit/833e9c1569b38fba9ea2f39d552c80496bca8472